src/dataStreams/DataStream.js

```diff
@@ -29,7 +34,7 @@ class DataStream {
     this.sourceSlotName = sourceSlotName
     this.sinkFctId = sinkFctId
     this.sinkSlotName = sinkSlotName
-    this.averagingWindowSize = averagingWindowSize
+    this.averagingWindowSize = averagingWindowSize || 0
```

Comment:
this was a mistake I found and fixed everywhere in this PR. default values are set by the constructor. The Joi schema is only used as validation here!